### PR TITLE
Added MySQL Native support

### PR DESCRIPTION
**Pull request by Gogen120**
- Added mysql_native datastore type to mysql resource `selectel_dbaas_mysql_datastore_v1`
- Added tests for MySQL Native

### CODE REVIEW

**Review by TawR1024 — CHANGES_REQUESTED** [review]

File: selectel/dbaas.go [hunk 1]
```diff
@@ -419,13 +426,34 @@ func resizeDatastore(ctx context.Context, d *schema.ResourceData, client *dbaas.
...
+	for _, dt := range expectedTypes {
+		if dt == datastoreType {
+			return true
+		}
+	}
+
+	return false
+}
+
+func buildDatastoreTypeErrorMEssage(expectedDatastoreTypeEngines []string, datastoreTypeEngine string) string {
```
[the remark this answers is not in the document]
Comment (Gogen120) [reply selectel/dbaas.go]: Fixed

File: selectel/dbaas.go [hunk 2]
```diff
@@ -419,13 +426,34 @@ func resizeDatastore(ctx context.Context, d *schema.ResourceData, client *dbaas.
 	return nil
 }
 
-func validateDatastoreType(ctx context.Context, expectedDatastoreTypeEngine string, typeID string, client *dbaas.API) diag.Diagnostics {
+func containeDatastoreType(expectedTypes []string, datastoreType string) bool {
+	for _, dt := range expectedTypes {
```
[the remark this answers is not in the document]
Comment (Gogen120) [reply selectel/dbaas.go]: Changed it to `expectedType`

File: selectel/dbaas.go [hunk 3]
```diff
@@ -420,13 +427,34 @@ func resizeDatastore(ctx context.Context, d *schema.ResourceData, client *dbaas.
...
+		baseMessage = "Provided datastore type must have an engine "
+	}
+
+	return baseMessage + strings.Join(expectedDatastoreTypeEngines, ", ") + " for this resource. But provided type is " + datastoreTypeEngine
+}
+
+func validateDatastoreType(ctx context.Context, expectedDatastoreTypeEngines []string, typeID string, client *dbaas.API) diag.Diagnostics {
 	datastoreType, err := client.DatastoreType(ctx, typeID)
 	if err != nil {
 		return diag.FromErr(errors.New("Couldnt get datastore type with id" + typeID))
```
Comment (TawR1024) [inline selectel/dbaas.go:454]: Maybe we should add space between message and `typeID`

Comment (Gogen120) [reply selectel/dbaas.go]: Fixed

**Review by TawR1024 — APPROVED** [review]

**Review by kolsean — APPROVED** [review]

**Pull request merged**